MAUI/Autocomplete/Searching-Filtering.md

```diff
@@ -171,6 +171,276 @@ autocomplete.TextSearchMode = AutocompleteTextSearchMode.Contains;
+
+{% tabs %}
+{% highlight c# %}
+
+//Model.cs
+
+public class CityInfo
+{
+    private bool _isCapital;
+    public bool IsCapital
+    {
+        get { return _isCapital; }
+        set { _isCapital = value; }
+    }
+    private string _cityname;
+    public string CityName
+    {
+        get { return _cityname; }
+        set { _cityname = value; }
+    }
+    private string _country;
+    public string CountryName
+    {
+        get { return _country; }
+        set { _country = value; }
+    }
+}
+
+//ViewModel.cs
+
+public class CityViewModel
+{
+    private ObservableCollection<CityInfo> _cityCollection;
+    public ObservableCollection<CityInfo> Cities
+    {
+        get { return _cityCollection; }
+        set { _cityCollection = value; }
+    }
+    public CityViewModel()
+    {
+        _cityCollection = new ObservableCollection<CityInfo>();
+
+        Cities.Add(new CityInfo() { CityName = "Delhi", CountryName = "India", IsCapital = true });
+        Cities.Add(new CityInfo() { CityName = "Mumbai", CountryName = "India", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "Chennai", CountryName = "India", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "Andhra Pradesh", CountryName = "India", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "Kerala", CountryName = "India", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "Chicago", CountryName = "USA", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "Los Angels", CountryName = "USA", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "Houston", CountryName = "USA", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "New York", CountryName = "USA", IsCapital = true });
+        Cities.Add(new CityInfo() { CityName = "Wasington", CountryName = "USA", IsCapital = false });
+        
+    }
+}
+{% endhighlight %}
+
+{% endtabs %}
+
+Now, let us create a custom filtering class to apply your filter logic to `Autocomplete` control by following these steps:
+
+**Step 1:** Create a class that derives from the `IAutocompleteFilterBehavior` interface.
+
+{% tabs %}
+{% highlight C# %}
+
+/// <summary>
+/// Represents a custom filtering behavior for `Autocomplete` control. 
+/// </summary>
+public class CityFilteringBehavior : IAutocompleteFilterBehavior
+{
+
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+**Step 2:** Then, implement the `GetMatchingIndexes` method of `IAutocompleteFilterBehavior` interface to create your own suggestion list (containing the indices of the filtered items) based on the text entered in the `Autocomplete` control that needs to be shown in the drop-down. The `GetMatchingIndexes` method contains following arguments:
```

Comment:
**Step 2:** Then, implement the `GetMatchingIndexes` method of the `IAutocompleteFilterBehavior` interface to create your suggestion list (containing the indices of the filtered items) based on the text entered in the `Autocomplete` control that needs to be shown in the drop-down. The `GetMatchingIndexes` method contains the following arguments.